### ISSUE TITLE
Python 3 stack overflow on logging errors

### ISSUE THREAD

**Issue by pevogam** [avocado-framework/avocado#3178]
Is there a very good reason why this function is in avocado's output module?:

```python
    def fake_outputs(self):
        """
        Replace sys.stdout/sys.stderr with in-memory-objects
        """
        sys.stdout = _StdOutputFile(True, self.records)
        sys.stderr = _StdOutputFile(False, self.records)
```

My problem with this is that it replaces the standard outputs where the standard logging module, unaware of such thing, decides to print errors encountered during
logging to `sys.stderr`. This then results in avocado handling the to-be-logged error
but it passes the ball back to the logging module which passes it back to stout/stderr. The final result is a recursion error but even worse yet, a stack overflow in the python interpreter because this change

```diff
diff --git a/Lib/logging/__init__.py b/Lib/logging/__init__.py
index b4659af7cc..7457549cb9 100644
--- a/Lib/logging/__init__.py
+++ b/Lib/logging/__init__.py
@@ -1094,6 +1094,8 @@ class StreamHandler(Handler):
             # issue 35046: merged two stream.writes into one.
             stream.write(msg + self.terminator)
             self.flush()
+        except RecursionError as e:
+            raise
         except Exception:
             self.handleError(record)
```

hasn't been deployed yet (even though it has been merged to python master). Ignoring the deeper python problem, I wonder how useful it is to replace very basic handles like the `stdout` and `stderr`. This could lead to many subtle errors, the one above just being an first-hand example.

**Comment by clebergnu** [comment]
@pevogam TBH problems keeping creeping up with the avocado.core.output and related functionality.  For instance:

* https://github.com/avocado-framework/avocado/issues/3187
* https://trello.com/c/j53LQn01/1064-bug-writing-to-stdout-fd-1-or-stderr-fd-2-should-not-affect-the-runner
* https://github.com/avocado-framework/avocado/pull/2131
* https://github.com/avocado-framework/avocado/pull/2090

So, the hard truth is that current Avocado does too much, and still not enough to (try to) isolate a test's output.  This is one of the reasons that has motivated a different approach altogether.

So, the (still rough and early) N(ext) Runner, attempts to solve this by *not* doing any magic.  A short example is here:
* https://github.com/avocado-framework/avocado/pull/3179/commits/46368fdf0a43afe9e482f9eade632eecb938f8dc#diff-5a2f92aedf41bef8d89fc436e1b3c84cR87

The produced output is simply transported is the test status message (which can be the final result).

**Comment by pevogam** [comment]
I see. Until we wait I can still propose a fix that stopped the issue from appearing (empirical confirmation only) at #3200.

**Comment by pevogam** [comment]
This issue is most probably fixed with the merge of #3200. Can reopen it if this proves wrong with time but the empirical probability is getting lower with every next test run.
